python/tvm/topi/arm_cpu/mprofile/dsp/micro_kernel/multi_channel_convolve.py

```diff
@@ -179,20 +179,26 @@ def _dual_int16_channel_convolve_impl(_tensor_h, tensor_w, channels, kernel_h, k
         extern "C"
         #endif
         int32_t {_get_func_name("int16", tensor_w, channels, kernel_h, kernel_w, suffix)}(
-            uint32_t *out,
-            uint32_t *tensor,
-            uint32_t *kernel) {{
+            int32_t *out,
+            int16_t *tensor,
+            int16_t *kernel) {{
 
-          uint32_t sum_c0 = 0;
-          uint32_t sum_c1 = 0;
+          int32_t sum_c0 = 0;
+          int32_t sum_c1 = 0;
+
+          int32_t kernel_i32[{kernel_h} * {kernel_w}];
+          memcpy(kernel_i32, kernel, {kernel_h} * {kernel_w} * 4);
+
+          int32_t tensor_length = {((kernel_w - 1) * (channels // 2) + (kernel_h - 1) * tensor_w * (channels // 2)) + 1};
```

Comment:
Curious why are braces used in this piece of logic?

Reply:
Because here the `tensor_length` value is computed in Python, do we prefer to do it in C?

Reply:
The formulae don't look wrong. But it does not look like a C syntax. So, if possible imho, we should drop braces.

Reply:
As discussion offline, I understand this piece now. The code is still in python.

Reply:
Yep, I agree this formula is correct.